File: ClientApp/app/models/student.ts
export interface IKeyValuePair {
  id: number;
  name: string;
}

export interface IClass {
  id: number;
  name: string;
  classSections: IKeyValuePair[];
}

export interface IAddress {
  street: string;
  city: string;
  country: string;
  zip: string;
}

export interface IStudent {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  sex: string;
  idNumber: string;
  dateOfBirth: string;
  dateEnrolled: string;
  class: IKeyValuePair;
  classSection: IKeyValuePair;
  address: IAddress;
}

export interface ISaveStudent {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  idNumber: string;
  sex: string;
  dateOfBirth: string;
  dateEnrolled: string;
  classId: number;
  classSectionId: number;
  address: IAddress;
}

File: Core/Models/Student.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace MySchool.Core.Models
{
    public class Student
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Sex { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string IdNumber { get; set; }
        public DateTime DateOfBirth { get; set; }
        public DateTime DateEnrolled { get; set; }
        public Address Address { get; set; }
        public int ClassSectionId { get; set; }
        public ClassSection ClassSection { get; set; }
        public ICollection<Photo> Photos { get; set; }

        public Student()
        {
            Photos = new Collection<Photo>();
        }
    }
}
